chain-storage/src/api.rs

```diff
@@ -0,0 +1,161 @@
+use std::collections::BTreeMap;
+
+use bit_vec::BitVec;
+use parity_scale_codec::Encode;
+
+use chain_core::common::H256;
+use chain_core::state::tendermint::BlockHeight;
+use chain_core::tx::data::{
+    input::{TxoIndex, TxoPointer},
+    TxId,
+};
+
+use super::buffer::{GetKV, StoreKV};
+use super::{
+    LookupItem, StoredChainState, CHAIN_ID_KEY, COL_APP_HASHS, COL_APP_STATES, COL_EXTRA,
+    COL_NODE_INFO, GENESIS_APP_HASH_KEY, LAST_STATE_KEY,
+};
+
+pub fn get_last_app_state(db: &impl GetKV) -> Option<Vec<u8>> {
```

Comment:
do all these need to be `pub` or are they expected only within the crate (`pub(super)` or `pub(crate)`)?

Reply:
These are the public api directly used by outside, because they can operate on different buffers.
Currently, the methods left in `Storage` are:
- read only, skipped the buffer, implies read committed semantic.
- genesis store methods, they are used before `ChainNodeApp` get initiated.

Reply:
it's more that similar undocumented public APIs are exposed both on "Storage" and here -- later when it's more cleaned up, it'll be good to consolidate it and document